Sources/Testing/Running/Runner.Context.swift

```diff
@@ -0,0 +1,226 @@
+  /// as the current configuration for a task.
+  ///
+  /// This property is used when an event is posted in a context where the value
+  /// of ``Configuration/current`` is `nil`, such as from a detached task.
+  static var all: some Collection<Self> {
+    _all.instances.values
+  }
+
+  /// Add this instance to ``Configuration/all``.
+  ///
+  /// - Returns: A unique number identifying `self` that can be
+  ///   passed to `_removeFromAll(identifiedBy:)`` to unregister it.
+  private func _addToAll() -> UInt64 {
+    Self.$_all.withLock { all in
+      let id = all.nextID
+      all.nextID += 1
+      all.instances[id] = self
+      return id
+    }
+  }
+
+  /// Remove this instance from ``Configuration/all``.
+  ///
+  /// - Parameters:
+  ///   - id: The unique identifier of this instance, as previously returned by
+  ///     `_addToAll()`. If `nil`, this function has no effect.
+  private func _removeFromAll(identifiedBy id: UInt64?) {
+    if let id {
+      Self.$_all.withLock { all in
+        _ = all.instances.removeValue(forKey: id)
+      }
+    }
+  }
+}
+
+// MARK: - Current test and test case
+
+extension Test {
+  /// The test that is running on the current task, if any.
+  public static var current: Self? {
+    Runner.Context.current.test
+  }
+
+  /// Call a function while the value of ``Test/current`` is set.
+  ///
+  /// - Parameters:
+  ///   - test: The new value to set for ``Test/current``.
+  ///   - body: A function to call.
+  ///
+  /// - Returns: Whatever is returned by `body`.
+  ///
+  /// - Throws: Whatever is thrown by `body`.
+  static func withCurrent<R>(_ test: Self, perform body: () throws -> R) rethrows -> R {
+    var context = Runner.Context.current
+    context.test = test
+    return try Runner.Context.$current.withValue(context, operation: body)
+  }
+
+  /// Call a function while the value of ``Test/current`` is set.
+  ///
+  /// - Parameters:
+  ///   - test: The new value to set for ``Test/current``.
+  ///   - body: A function to call.
+  ///
+  /// - Returns: Whatever is returned by `body`.
+  ///
+  /// - Throws: Whatever is thrown by `body`.
+  static func withCurrent<R>(_ test: Self, perform body: () async throws -> R) async rethrows -> R {
+    var context = Runner.Context.current
+    context.test = test
+    return try await Runner.Context.$current.withValue(context, operation: body)
+  }
+}
+
+extension Test.Case {
+  /// The test case that is running on the current task, if any.
+  public static var current: Self? {
+    Runner.Context.current.testCase
+  }
+
+  /// Call a function while the value of ``Test/Case/current`` is set.
+  ///
+  /// - Parameters:
+  ///   - testCase: The new value to set for ``Test/Case/current``.
+  ///   - body: A function to call.
+  ///
+  /// - Returns: Whatever is returned by `body`.
+  ///
+  /// - Throws: Whatever is thrown by `body`.
+  static func withCurrent<R>(_ testCase: Self, perform body: () throws -> R) rethrows -> R {
```

Comment:
Do we use the `sync` variants? If they're just dead code, let's drop 'em.

Reply:
Ah yes, we do not need them, and removing them improves code coverage.